dbms/src/Common/UInt128.h

```diff
@@ -25,19 +24,43 @@ struct UInt128
     UInt64 first;
     UInt64 second;
 
-    bool operator== (const UInt128 rhs) const { return first == rhs.first && second == rhs.second; }
-    bool operator!= (const UInt128 rhs) const { return first != rhs.first || second != rhs.second; }
-
-    bool operator== (const UInt64 rhs) const { return first == rhs && second == 0; }
-    bool operator!= (const UInt64 rhs) const { return first != rhs || second != 0; }
+    UInt128() = default;
+    explicit UInt128(const UInt64 rhs) : first(0), second(rhs) { }
+
+    bool operator== (const UInt128 & rhs) const { return second == rhs.second && first == rhs.first; }
```

Comment:
Almost no reason to pass by reference here (and also no harm as compiler will remove reference as one of optimization passes) - it's just about style preferences.